arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc

```diff
@@ -5,13 +5,177 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+config SOC_NRF52810
+	depends on SOC_SERIES_NRF52X
+	bool
+	select HAS_HW_NRF_CCM
+	select HAS_HW_NRF_CLOCK
+	select HAS_HW_NRF_COMP
+	select HAS_HW_NRF_ECB
+	select HAS_HW_NRF_EGU0
+	select HAS_HW_NRF_EGU1
+	select HAS_HW_NRF_GPIO0
+	select HAS_HW_NRF_GPIOTE
+	select HAS_HW_NRF_PDM
+	select HAS_HW_NRF_POWER
+	select HAS_HW_NRF_PPI
+	select HAS_HW_NRF_PWM0
+	select HAS_HW_NRF_QDEC
+	select HAS_HW_NRF_RNG
+	select HAS_HW_NRF_RTC0
+	select HAS_HW_NRF_RTC1
+	select HAS_HW_NRF_SAADC
+	select HAS_HW_NRF_SPIM0
+	select HAS_HW_NRF_SPIS0
+	select HAS_HW_NRF_SWI0
+	select HAS_HW_NRF_SWI1
+	select HAS_HW_NRF_SWI2
+	select HAS_HW_NRF_SWI3
+	select HAS_HW_NRF_SWI4
+	select HAS_HW_NRF_SWI5
+	select HAS_HW_NRF_TEMP
+	select HAS_HW_NRF_TIMER0
+	select HAS_HW_NRF_TIMER1
+	select HAS_HW_NRF_TIMER2
+	select HAS_HW_NRF_TWIM0
+	select HAS_HW_NRF_TWIS0
+	select HAS_HW_NRF_UARTE0
+	select HAS_HW_NRF_WDT
+
 config SOC_NRF52832
 	depends on SOC_SERIES_NRF52X
 	bool
+	select CPU_HAS_FPU
+	select HAS_HW_NRF_CCM
+	select HAS_HW_NRF_CLOCK
+	select HAS_HW_NRF_COMP
+	select HAS_HW_NRF_ECB
+	select HAS_HW_NRF_EGU0
+	select HAS_HW_NRF_EGU1
+	select HAS_HW_NRF_EGU2
+	select HAS_HW_NRF_EGU3
+	select HAS_HW_NRF_EGU4
+	select HAS_HW_NRF_EGU5
+	select HAS_HW_NRF_GPIO0
+	select HAS_HW_NRF_GPIOTE
+	select HAS_HW_NRF_I2S
+	select HAS_HW_NRF_LPCOMP
+	select HAS_HW_NRF_MWU
+	select HAS_HW_NRF_NFCT
+	select HAS_HW_NRF_PDM
+	select HAS_HW_NRF_POWER
+	select HAS_HW_NRF_PPI
+	select HAS_HW_NRF_PWM0
+	select HAS_HW_NRF_PWM1
+	select HAS_HW_NRF_PWM2
+	select HAS_HW_NRF_QDEC
+	select HAS_HW_NRF_RNG
+	select HAS_HW_NRF_RTC0
+	select HAS_HW_NRF_RTC1
+	select HAS_HW_NRF_RTC2
+	select HAS_HW_NRF_SAADC
+	select HAS_HW_NRF_SPI0
+	select HAS_HW_NRF_SPI1
+	select HAS_HW_NRF_SPI2
+	select HAS_HW_NRF_SPIM0
+	select HAS_HW_NRF_SPIM1
+	select HAS_HW_NRF_SPIM2
+	select HAS_HW_NRF_SPIS0
+	select HAS_HW_NRF_SPIS1
+	select HAS_HW_NRF_SPIS2
+	select HAS_HW_NRF_SWI0
+	select HAS_HW_NRF_SWI1
+	select HAS_HW_NRF_SWI2
+	select HAS_HW_NRF_SWI3
+	select HAS_HW_NRF_SWI4
+	select HAS_HW_NRF_SWI5
+	select HAS_HW_NRF_TEMP
+	select HAS_HW_NRF_TIMER0
+	select HAS_HW_NRF_TIMER1
+	select HAS_HW_NRF_TIMER2
+	select HAS_HW_NRF_TIMER3
+	select HAS_HW_NRF_TIMER4
+	select HAS_HW_NRF_TWI0
+	select HAS_HW_NRF_TWI1
+	select HAS_HW_NRF_TWIM0
+	select HAS_HW_NRF_TWIM1
+	select HAS_HW_NRF_TWIS0
+	select HAS_HW_NRF_TWIS1
+	select HAS_HW_NRF_UART0
+	select HAS_HW_NRF_UARTE0
+	select HAS_HW_NRF_WDT
 
 config SOC_NRF52840
 	depends on SOC_SERIES_NRF52X
 	bool
+	select CPU_HAS_FPU
+	select XIP
```

Comment:
Why XIP is not selected in nRF52832?

Reply:
fixed
I was told that this is different XIP. I've created one more configuration parameter:
HAS_HW_NRF_XIP.